user_unix.go

```diff
@@ -0,0 +1,68 @@
+// +build linux darwin
+
+package lifecycle
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"strconv"
+	"syscall"
+)
+
+var currentUser = os.Getuid()
+
+// asUser sets HOME and non-root user credentials on command
+func asUser(cmd *exec.Cmd, uid, gid int) (*exec.Cmd, error) {
+	if currentUser != 0 {
+		cmd.Env = append(cmd.Env, "HOME="+os.Getenv("HOME"))
+		return cmd, nil
+	}
+	user, err := user.LookupId(strconv.Itoa(uid))
+	if err != nil {
+		return nil, err
+	}
+	cmd.Env = append(cmd.Env, "HOME="+user.HomeDir)
+	cmd.SysProcAttr = &syscall.SysProcAttr{
+		Credential: &syscall.Credential{
+			Uid: uint32(uid),
+			Gid: uint32(gid),
+		},
+	}
+	return cmd, nil
+}
+
+func ensureOwner(path string, uid, gid int) error {
```

Comment:
This method behaves unexpected if you aren't root (by failing to ensure ownership and not returning an error).